packages/big-design/src/components/Textarea/Textarea.tsx

```diff
@@ -21,88 +21,105 @@ interface PrivateProps {
+        </FormControlLabel>
+      );
+    }
 
-  private getId() {
-    const { id } = this.props;
+    if (React.isValidElement(label) && label.type === FormControlLabel) {
+      return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
+        id: labelId,
+        htmlFor: id,
+      });
+    }
 
-    return id ? id : this.uniqueId;
-  }
+    if (!label) {
+      return null;
+    }
 
-  private renderDescription() {
-    const { description } = this.props;
+    warning('label must be either a string or a FormControlLabel component.');
+  }, [label, labelId, props.required]);
 
+  const renderedDescription = useMemo(() => {
     if (typeof description === 'string') {
-      return <Textarea.Description>{description}</Textarea.Description>;
+      return <FormControlDescription>{description}</FormControlDescription>;
     }
 
-    if (React.isValidElement(description) && description.type === Textarea.Description) {
+    if (React.isValidElement(description) && description.type === FormControlDescription) {
       return description;
     }
 
-    return null;
-  }
+    if (!description) {
+      return null;
+    }
 
-  private renderLabel() {
-    const { label, labelId, required } = this.props;
-    const id = this.getId();
+    warning('description must be either a string or a FormControlDescription component.');
+  }, [description]);
 
-    if (typeof label === 'string') {
-      return (
-        <Textarea.Label id={labelId} htmlFor={id} renderOptional={!required}>
-          {label}
-        </Textarea.Label>
-      );
+  const numOfRows = useMemo(() => {
+    if (rows && rows > MAX_ROWS) {
+      return MAX_ROWS;
     }
 
-    if (React.isValidElement(label) && label.type === Textarea.Label) {
-      return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
-        id: labelId,
-        htmlFor: id,
-      });
+    return rows;
+  }, [rows]);
+
+  const validateError = (err: Props['error']) => {
+    if (typeof err === 'string') {
+      return err;
     }
 
-    return null;
-  }
+    if (React.isValidElement(err) && err.type === FormControlError) {
+      return err;
+    }
 
-  private getRows(rows: Props['rows']) {
-    if (rows && rows > this.MAX_ROWS) {
-      return this.MAX_ROWS;
+    if (!err) {
+      return null;
     }
 
-    return rows;
-  }
-}
+    warning('error must be either a string or a FormControlError component.');
+  };
 
-const TextareaWithForwardedRef = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
-  ({ className, style, ...props }, ref) => <StyleableTextarea {...props} forwardedRef={ref} />,
-);
+  const errors = useMemo(() => {
+    if (Array.isArray(error)) {
+      error.forEach(validateError);
 
-export const Textarea = hoistNonReactStatics(TextareaWithForwardedRef, StyleableTextarea);
+      return error;
+    }
 
-Textarea.displayName = 'Textarea';
+    return validateError(error);
+  }, [error]);
```

Comment:
Missing `validateError` dependency, consider moving it inside this memo if not being used elsewhere.